docs/reference/robot.md

```diff
@@ -240,6 +252,22 @@ It means that the step actually lasted the requested number of milliseconds, but
 - if `dt` > `duration`, then the actuators values were set at `controller_time` + `dt`, and the sensor values were also measured at `controller_time` + `dt`.
 It means that the requested step duration could not be respected.
 
+When using `wb_robot_step`, the controller code is executed sequentially with the Webots simulation step, i.e., not in parallel.
+This is due to the fact that a typical controller reads sensor information, makes some computation, orders motor commands and calls `wb_robot_step` which actually sends the motor commands and sensors requests and waits until Webots completes a simulation step, which may take some time depending on the complexity of the simulation.
+During this time, the controller is idle, waiting for Webots to complete its simulation step.
+On the other hand, prior to starting a new step, Webots waits for all the controllers to send their `wb_robot_step` messages which may induce some idle waiting time in Webots if a controller doesn't send quickly enough its `wb_robot_step` message because it is busy with some computation.
+If the two computational processes (Webots and controller) are slow, it may be interesting to parallelize them.
+`wb_robot_step_begin` and `wb_robot_step_end` allow you to achieve such an implementation.
+They correspond to a split version of `wb_robot_step`, with the particularity that the code written between the two function calls is executed in parallel with the Webots simulation step.
+`wb_robot_step_begin` and `wb_robot_step_end` both return -1 if the simulation is terminated.
+Note that some functions cannot be implemented between `wb_robot_step_begin` and `wb_robot_step_end`, as they require immediate response from Webots using a request-response pattern. 
+This includes some [Supervisor API](supervisor.md) functions, like `wb_supervisor_node_get_field` and `wb_supervisor_field_get_sf_rotation`, but also some Robot API functions, like `wb_robot_get_urdf`. 
+Webots warns you in case you implement one of these functions between `wb_robot_step_begin` and `wb_robot_step_end`.
+You can simply write them before `wb_robot_step_begin` or after `wb_robot_step_end`.
+If you still want to parallelize some of these functions with the simulation step, you can use the tracking options from the supervisor. 
+`wb_supervisor_field_enable_sf_tracking`, `wb_supervisor_node_enable_pose_tracking` and `wb_supervisor_node_enable_contact_point_tracking` force Webots to stream information to the controller. 
+Using tracking, some supervisor methods get their value locally and can be written between `wb_robot_step_begin` and `wb_robot_step_end`.
```

Comment:
```suggestion
By enabling the tracking, the corresponding supervisor functions can be called between `wb_robot_step_begin` and `wb_robot_step_end`, because their value will be queried to Webots during `wb_robot_step_begin` and received during `wb_robot_step_end`.
```